std/algorithm/searching.d

```diff
@@ -4353,6 +4354,19 @@ template skipOver(alias pred = (a, b) => a == b)
     assert(s2.skipOver("", "foo"));
 }
 
+// dxml regression
+@safe unittest
+{
+    import std.utf : byCodeUnit;
+
+    bool stripStartsWith(Text)(ref Text text, string needle)
+    {
+        return text.skipOver(needle.byCodeUnit());
+    }
+    auto text = "<xml></xml>"d.byCodeUnit;
+    assert(stripStartsWith(text, "<xml>"));
```

Comment:
It seems like you should verify that the right characters were skipped.

Reply:
Done. The error was an compilation error, but added the verification anyhow as it doesn't hurt.